python/tests/test_collections.py

```diff
@@ -0,0 +1,68 @@
+from unittest.mock import Mock
+
+import pytest
+from numerous import collection
+from numerous._client import Client
+from numerous.generated.graphql.client import Client as GQLCleint
+from numerous.generated.graphql.fragments import CollectionNotFound, CollectionReference
+
+
+ORGANIZATION_ID = "test_org"
+COLLECTION_NAME = "test_collection"
+NESTED_COLLECTION_NAME = "nested_test_collection"
+COLLECTION_REFERENCE = CollectionReference(key="test_key", id="test_id")
+NESTED_COLLECTION_REFERENCE=CollectionReference(key="nested_test_key",
+                                                 id="nested_test_id")
+
+
+
+@pytest.fixture(autouse=True)
+def _set_env_vars(monkeypatch:pytest.MonkeyPatch)->None:
+    monkeypatch.setenv("NUMEROUS_API_URL", "url_value")
+    monkeypatch.setenv("NUMEROUS_API_ACCESS_TOKEN", "token")
+
+
+def test_collection_new_key()->None:
+    gql = Mock(GQLCleint)
+    _client = Client(gql)
+    gql.collection_create.return_value = Mock(collection_create = COLLECTION_REFERENCE)
```

Comment:
I think the mocked method should return the proper values. Add some helper methods to create the proper values that `gql.collection_create` would actually return (even though they have long names it should be OK :grin:)